routing/routing_integration_tests/bicycle_route_test.cpp

```diff
@@ -241,87 +330,123 @@ UNIT_TEST(Spain_Madrid_ElevationsDetour)
 
-UNIT_TEST(Stolbcy_Use_Unpaved)
+UNIT_TEST(Belarus_Stolbcy_Use_Unpaved)
 {
-  // Same as GraphHopper. Valhalla uses ground path through ford.
-  // OSRM makes completely strange route.
+  // Goes by unpaved and paved streets and an unpaved_bad track in the end.
+  // Same as GraphHopper. Valhalla detours the unpaved street.
+  // OSRM shortcuts through paths and a ford.
   CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
       mercator::FromLatLon(53.471389, 26.7422186), {0.0, 0.0},
       mercator::FromLatLon(53.454082, 26.7560061), 5148.45 /* expectedRouteMeters */);
 }
 
-UNIT_TEST(Russia_UseTrunk)
+UNIT_TEST(Russia_UseTrunk_Long)
 {
-  // Similar with GraphHopper.
+  // Prefer riding via a straight trunk road instead of taking a long +30% detour via smaller roads.
+  /// @todo(pastk) This test is controversial as this route has sections with longer relative trunk-avoiding detours
+  /// e.g. from 67.9651692,32.8685132 to 68.022093,32.9654391 its +40% longer via a teriary and its OK?
   CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
                                    mercator::FromLatLon(66.271, 33.048), {0.0, 0.0},
                                    mercator::FromLatLon(68.95, 33.045), 412795 /* expectedRouteMeters */);
 }
 
-// https://github.com/organicmaps/organicmaps/issues/3920
-UNIT_TEST(IgnoreCycleBarrier_WithoutAccess)
+/// @todo(pastk): find a good "avoid trunk" test case.
+UNIT_TEST(Russia_UseTrunk)
 {
+  // Prefer riding via a straight trunk road instead of taking weird detours via smaller roads
+  // (651m via a tertiaty + service in this case).
   CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
-                                   mercator::FromLatLon(51.9960994, 5.67350176), {0.0, 0.0},
-                                   mercator::FromLatLon(51.9996861, 5.67299133), 428.801);
+                                   mercator::FromLatLon(68.0192918, 32.9576269), {0.0, 0.0},
+                                   mercator::FromLatLon(68.0212968, 32.9632167), 323.951 /* expectedRouteMeters */);
+}
 
-  // The difference here (end points are almost equal) because of some barrier=gate without access.
-  // https://www.openstreetmap.org/node/6993853766
+UNIT_TEST(Russia_UseTrunk_AvoidGasStationsDetours)
+{
+  /// @todo(pastk): still OM detours into many petrol stations along the trunk.
+  CalculateRouteAndTestRouteLength(GetVehicleComponents(VehicleType::Bicycle),
+                                   mercator::FromLatLon(55.9132468, 39.1453188), {0.0, 0.0},
+                                   mercator::FromLatLon(55.9146268, 39.153307), 549.911 /* expectedRouteMeters */);
+}
 
-  /// @todo Make correct tunnels elevation (linear between start and finish).
+UNIT_TEST(Netherlands_IgnoreCycleBarrier_WithoutAccess)
+{
+  // There is a barrier=cycle_barrier in the beginning of the route
+  // and it doesn't affect routing as there is no explicit bicycle=no.
```

Comment:
```suggestion
  // and it doesn't affect routing as there is no explicit bicycle=no.
  /// @todo(pastk): such barrier should have a small penalty in routing.
```

Reply:
one more cases where penalty for points is needed, though tine as these are often on preferred cycling routes

Reply:
good point!